Added title for the remove liquidity modal

valeryV99 opened the pull request:
Added title for the remove liquidity modal

https://www.notion.so/254362757c98814193bdf2683a2f8535?v=254362757c98811fb5a9000c47b44575&p=25d362757c9880038e94ccb121824782&pm=s

valeryV99 committed:
feat: enhance RemoveLiquidityDialog with improved layout and styling

- Added header to guide users during the intention signing process.
- Introduced vertical stack for better alignment and spacing.

valeryV99 committed:
feat(tests): add unit tests for RemoveLiquidityDialog and update Vitest config

- Added comprehensive unit tests for `RemoveLiquidityDialog` to improve test coverage.
- Enhanced Vitest configuration to include the new `RemoveLiquidityDialog` test files.
- Updated dependencies in `pnpm-lock.yaml` to align with required test tools and features for the latest Vitest version.

valeryV99 committed:
test: remove redundant comments in RemoveLiquidityDialog tests

- Cleaned up unused and redundant comments to improve test code readability.
- Simplified test logic while maintaining functionality and coverage.

valeryV99 committed:
test: adjust indentation in RemoveLiquidityDialog tests

- Fixed inconsistent indentation to improve readability and maintain coding standards in `RemoveLiquidityDialog.test.tsx`.

valeryV99 committed:
test: fix indentation and improve readability in RemoveLiquidityDialog tests

- Adjusted inconsistent indentation to adhere to coding standards.
- Refactored test structure for improved clarity and maintainability.

opcode0x commented on apps/web/src/features/liquidity/ui/remove-liquidity-dialog/RemoveLiquidityDialog.test.tsx:
```diff
@@ -0,0 +1,248 @@
+
+import { Address, parseUnits, zeroAddress } from 'viem';
+
+vi.mock('@midl-xyz/satoshi-kit', async () => ({
```
Why do we need to mock this modules?

valeryV99 replied:
Yes
Without this mock, the tests fail

valeryV99 replied:
done

opcode0x commented on apps/web/src/features/liquidity/ui/remove-liquidity-dialog/RemoveLiquidityDialog.test.tsx:
```diff
@@ -0,0 +1,248 @@
+  useSlippage: () => [0.01], // 1%
+}));
+
+vi.mock('@/features/token', () => ({
```
Do we need to mock this component?

valeryV99 replied:
removed

opcode0x commented on apps/web/src/features/liquidity/ui/remove-liquidity-dialog/RemoveLiquidityDialog.test.tsx:
```diff
@@ -0,0 +1,248 @@
+    useToken: () =>
+      mockRunePresent ? { rune: { id: 'rune-1' } } : { rune: undefined },
+    useERC20Rune: () => ({ rune: { id: 'rune-1', symbol: 'AAA' } }),
+  } as unknown as typeof original;
```
as far as i know if you follow the typings it should work without typecasting

valeryV99 replied:
`+`

opcode0x commented on apps/web/src/features/liquidity/ui/remove-liquidity-dialog/RemoveLiquidityDialog.test.tsx:
```diff
@@ -0,0 +1,248 @@
+  useChainId: () => 1,
+}));
+
+vi.mock('@/features/btc/ui/IntentionSigner', () => ({
```
don't mock component just to add test-id

valeryV99 replied:
`+`

valeryV99 committed:
test: update mocks and refactor RemoveLiquidityDialog tests

- Removed redundant mocks from `RemoveLiquidityDialog.test.tsx` for improved clarity and maintainability.
- Added new mock implementations for `useAddTxIntention`, `useFinalizeBTCTransaction`, and related hooks to enhance test coverage.
- Adjusted `IntentionSigner` related assertions to focus on rendering validation instead of internal attributes.
- Fixed import path for `TokenLogo` in `TokenValue` component.
- Added `data-testid` attribute to `IntentionSigner` for more consistent test targeting.

valeryV99 committed:
Merge remote-tracking branch 'origin/staging' into fix/swap-13

valeryV99 committed:
test: refactor RemoveLiquidityDialog tests and improve dialog layout

- Replaced `QueryClientProvider` wrapper with streamlined `Wrapper` component in `RemoveLiquidityDialog.test.tsx`.
- Removed unused mocks and redundant test variables for clarity and maintainability.
- Updated `DialogTitle` usage for better accessibility and semantic structure.
- Added a “Close” button for enhanced user interaction and testing.

valeryV99 committed:
test: streamline RemoveLiquidityDialog tests and update Vitest config

- Removed unnecessary `cleanup()` invocation in `RemoveLiquidityDialog.test.tsx` setup.
- Enabled global test variables in Vitest configuration for consistency.
- Applied minor formatting improvements in Vitest config for better readability.
